azure-devops/azext_devops/dev/common/telemetry.py

```diff
@@ -87,4 +89,30 @@ def _send_tracking_ci_event(devops_organization=None, ci_client=None):
         return False
 
 
+def _get_shell_type():
+    import os
+    if 'ZSH_VERSION' in os.environ:
+        return 'zsh'
+    if 'BASH_VERSION' in os.environ:
+        return 'bash'
+    if 'KSH_VERSION' in os.environ or 'FCEDIT' in os.environ:
+        return 'ksh'
+    if 'WINDIR' in os.environ:
+        return 'cmd'
+    return _remove_cmd_chars(_remove_symbols(os.environ.get('SHELL')))
```

Comment:
not able to get this..
can you put some comments here, for future readers as well